### ISSUE TITLE
Emit method overload symbol in Scala mtags

### ISSUE THREAD

**Issue by olafurpg** [scalameta/metals#282]
Java mtags currently emits field symbols for methods.
```scala
def foo(a: Int, b: String) 
// field: foo.
// method foo(Int, String)
```
it would be nice if Scala mtags emitted correct method symbols so it's possible to goto definition to the precise overload of a method.

The specification for Scala method symbol syntax is documented here https://github.com/scalameta/scalameta/blob/master/semanticdb/semanticdb3/semanticdb3.md#scala-symbol

The place to fix this issue is here https://github.com/scalameta/metals/blob/905bf282299f7c136b40173103cf367073dc55f8/metals/src/main/scala/scala/meta/metals/mtags/SclaMtags.scala
and to test the fix

https://github.com/scalameta/metals/blob/905bf282299f7c136b40173103cf367073dc55f8/metals/src/test/scala/tests/mtags/ScalaMtagsTest.scala

Related #281

**Comment by amarrella** [comment]
I'm gonna try to solve this today :)

**Comment by olafurpg** [comment]
Fixed in #292, mtags emits correct method overloads for both Scala and Java and we have a lot better tests to detect regressions in this space.
